client/src/Login.tsx

```diff
@@ -73,10 +83,18 @@ function Login() {
                 },
                 body: JSON.stringify(formData),
             });
-            const accessToken = await res.json()
-            history.push('/')
+            const data = await res.json()
+            if(data.status === 401){
+                setStatus('failure')
+                if(data.error === 'Email not found'){
+                    setError('email')
+                }
+                else if(data.error === 'Invalid password'){
+                    setError('password')
+                }
+            } else setStatus('success')
         } catch (err) {
-            // Handle error
+            setStatus('failure')
```

Comment:
I think if the server is down, this would catch that kind of failure on a request. You think we could handle that elsewhere. Or I guess if it is all served from one deployment then the frontend will not exist without the backend also existing.

Reply:
yeah, that last part. but i guess it doesn't hurt. up to you

Reply:
Removed the try-catch. Better to build it towards the end goal.